frappe/model/db_query.py

```diff
@@ -426,29 +431,10 @@ def prepare_filter_condition(self, f):
 			if df and df.fieldtype in ("Check", "Float", "Int", "Currency", "Percent"):
 				can_be_null = False
 
-			if f.operator.lower() in ('previous', 'next'):
-				if f.operator.lower() == "previous":
-					if f.value == "1 week":
-						date_range = [add_to_date(nowdate(), days=-7), nowdate()]
-					elif f.value == "1 month":
-						date_range = [add_to_date(nowdate(), months=-1), nowdate()]
-					elif f.value == "3 months":
-						date_range = [add_to_date(nowdate(), months=-3), nowdate()]
-					elif f.value == "6 months":
-						date_range = [add_to_date(nowdate(), months=-6), nowdate()]
-					elif f.value == "1 year":
-						date_range = [add_to_date(nowdate(), years=-1), nowdate()]
-				elif f.operator.lower() == "next":
-					if f.value == "1 week":
-						date_range = [nowdate(), add_to_date(nowdate(), days=7)]
-					elif f.value == "1 month":
-						date_range = [nowdate(), add_to_date(nowdate(), months=1)]
-					elif f.value == "3 months":
-						date_range = [nowdate(), add_to_date(nowdate(), months=3)]
-					elif f.value == "6 months":
-						date_range = [nowdate(), add_to_date(nowdate(), months=6)]
-					elif f.value == "1 year":
-						date_range = [nowdate(), add_to_date(nowdate(), years=1)]
+			if f.operator.lower() == 'previous' or f.operator.lower() == 'next':
+				f = replace_old_date_filters(f)
```

Comment:
Handle this on the client-side so that "previous" & "next" can be deprecated easily once all the user settings are updated.

"timespan" can be sent from client-side whenever "previous" or "next" is encountered & user_settings will be updated accordingly.

Reply:
@surajshetty3416 If we handle this on the client side, filters in charts, reports or anywhere else won't be handled

Reply:
Yeah, valid... so let's just not have "replace_old_date_filters" in the codebase, since there are no "old_date_filters" and we have to support "previous" and "next" anyway... so the code can be something like...
```suggestion
			if f.operator.lower() in ('previous', 'next', 'timespan'):
				date_range = get_date_range(f.value, f.operator.lower())
```